### DISCUSSION TITLE
How to render alternate language links for each route?

### DISCUSSION THREAD

**Discussion by Blaumaus** [remix-run/remix#8159] [Q&A]
I implemented alternate href links in a `root.jsx` file like that:
```javascript
const alternateLinks = _reduce(
    data?.allSites,
    (acc, {subdomain, countryCode}) => {
      if (!subdomain || !countryCode) {
        return acc;
      }

      const locale = clm.getLocaleByAlpha2(_toUpper(countryCode));
      const urlObject = new URL(subdomain);
      urlObject.pathname = pathname;

      return [
        ...acc,
        {
          rel: 'alternate',
          hrefLang: locale,
          href: urlObject.toString(),
        },
      ];
    },
    [],
  );
``` 

```html
<head>
  <Links />
  {_map(alternateLinks, (link) => (
    <link key={link.hrefLang} {...link} />
  ))}
  {/* todo: x-default hreflang link */}
</head>
```

it renders the links like it's supposed to, but the problem is if I navigate to another page - the alt links will stay the same because root loader executes once.

How do I properly render them for all pages?
Am I supposed to do something like that in each route or is it possible to use some kind of hook or something similar?

**Comment by sergiodxa** [comment]
You can use the MetaFunction to render links that depends on data or the url.
